src/goto-programs/graphml_witness.cpp

```diff
@@ -124,18 +188,57 @@ std::string graphml_witnesst::convert_assign_rec(
         break;
     }
   }
+  else if(assign.rhs().id() == ID_with)
+  {
+    const with_exprt &with_expr = to_with_expr(assign.rhs());
+    const auto &ops = with_expr.operands();
+
+    for(std::size_t i = 1; i < ops.size(); i += 2)
+    {
+      if(!result.empty())
+        result += ' ';
+
+      if(ops[i].id() == ID_member_name)
+      {
+        const member_exprt member{
+          assign.lhs(), ops[i].get(ID_component_name), ops[i + 1].type()};
+        result +=
+          convert_assign_rec(identifier, code_assignt(member, ops[i + 1]));
+      }
+      else
+      {
+        const index_exprt index{assign.lhs(), ops[i]};
```

Comment:
Invariant that this is an `ID_array_index` or whatever it's called?

Reply:
No, this is the literal index, not a wrapper around an index. It could be any expression.